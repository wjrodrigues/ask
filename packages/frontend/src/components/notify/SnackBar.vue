<template>
  <div class="text-center">
    <v-snackbar v-model="snackbar" multi-line :timeout="timeout">
      {{ text }}

      <template v-slot:actions>
        <v-btn rounded="pill" variant="text" @click="close()">
          {{ $t("notify.snackbar.close") }}
        </v-btn>
      </template>
    </v-snackbar>
  </div>
</template>

<script lang="ts">
import notify from "@/components/notify";

export default {
  data: () => ({
    snackbar: false,
    timeout: 5000,
    text: "",
  }),
  mounted() {
    notify.snackbar.listen((text: string) => {
      if (text.trim().length != 0) {
        this.text = text;
        this.snackbar = true;
      }
    });
  },
  methods: {
    close: function () {
      this.snackbar = false;
      this.text = "";
    },
  },
};
</script>
