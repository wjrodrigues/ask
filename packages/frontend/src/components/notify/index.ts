interface Notify {
  action: Function;
  listen: Function;
  call: Function;
}

const snackbar = {
  action: () => {},
  listen: function (func: Function) {
    this.action = func;
  },
  call: function (text: string) {
    this.action(text);
  },
} as Notify;

export default { snackbar };
