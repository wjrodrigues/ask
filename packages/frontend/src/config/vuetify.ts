import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { aliases, fa } from "vuetify/iconsets/fa";
import { createVuetify } from "vuetify";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";

const vuetify = createVuetify({
  components,
  directives,
  icons: {
    defaultSet: "fa",
    aliases,
    sets: {
      fa,
    },
  },
});

export default vuetify;
