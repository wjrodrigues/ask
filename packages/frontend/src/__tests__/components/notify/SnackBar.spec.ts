import { describe, it, expect } from "vitest";
import { mount, shallowMount } from "@vue/test-utils";

import SnackBar from "@/components/notify/SnackBar.vue";
import notify from "@/components/notify";
import { basic_mount } from "@/__tests__/help";
import { VBtn } from "vuetify/components";

describe("when render SnackBar", () => {
  it("render SnackBar with text", async () => {
    const wrapper = await shallowMount(
      SnackBar,
      basic_mount({ props: {}, plugins: [] })
    );
    await notify.snackbar.call("Vue App");
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.$data).toEqual({
      snackbar: true,
      timeout: 5000,
      text: "Vue App",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("does not render SnackBar when empty text", async () => {
    const wrapper = await shallowMount(
      SnackBar,
      basic_mount({ props: {}, plugins: [] })
    );
    await notify.snackbar.call("");
    await wrapper.vm.$nextTick();

    expect(wrapper.vm.$data).toEqual({
      snackbar: false,
      timeout: 5000,
      text: "",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("updates data if click the close button", async () => {
    const wrapper = await mount(
      SnackBar,
      basic_mount({ props: {}, plugins: [] })
    );

    await notify.snackbar.call("Vue App");
    expect(wrapper.vm.$data).toEqual({
      snackbar: true,
      timeout: 5000,
      text: "Vue App",
    });

    const btnClose = wrapper.findComponent(VBtn);
    await btnClose.trigger("click");

    expect(wrapper.vm.$data).toEqual({
      snackbar: false,
      timeout: 5000,
      text: "",
    });
  });
});
