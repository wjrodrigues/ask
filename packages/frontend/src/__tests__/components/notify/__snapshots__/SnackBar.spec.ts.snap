// Vitest Snapshot v1

exports[`when render SnackBar > does not render SnackBar when empty text 1`] = `
"<div class=\\"text-center\\">
  <v-snackbar-stub multiline=\\"true\\" timeout=\\"5000\\" vertical=\\"false\\" location=\\"bottom\\" variant=\\"elevated\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" modelvalue=\\"false\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" origin=\\"auto\\" transition=\\"v-snackbar-transition\\"></v-snackbar-stub>
</div>"
`;

exports[`when render SnackBar > render SnackBar with text 1`] = `
"<div class=\\"text-center\\">
  <v-snackbar-stub multiline=\\"true\\" timeout=\\"5000\\" vertical=\\"false\\" location=\\"bottom\\" variant=\\"elevated\\" absolute=\\"false\\" attach=\\"false\\" closeonback=\\"true\\" contained=\\"false\\" disabled=\\"false\\" modelvalue=\\"true\\" zindex=\\"2000\\" activatorprops=\\"[object Object]\\" openonhover=\\"false\\" closeoncontentclick=\\"false\\" eager=\\"false\\" locationstrategy=\\"static\\" origin=\\"auto\\" transition=\\"v-snackbar-transition\\"></v-snackbar-stub>
</div>"
`;
